Library/Homebrew/formula.rb

```diff
@@ -476,6 +476,29 @@ def compiler_failures
     active_spec.compiler_failures
   end
 
+  # The best spec we can upgrade the {Formula} to so it can satisfy dependency.
+  def best_upgrade_spec_sym
```

Comment:
Just to give my opinion as well, I believe the decision to stay on the HEAD spec is the correct one.